lib/repl.js

```diff
@@ -226,6 +229,66 @@ function REPLServer(prompt,
 
   const self = this;
 
+  self[kPreviewResults] = (eagerSession, eagerEvalContextId) => {
+    this.on('buffer', (line) => {
+      Interface.prototype._clearPreview.call(this);
+
+      // No need of preview for a multiline statement
+      if (this[kBufferedCommandSymbol] !== '')
+        return;
+
+      eagerSession.post('Runtime.evaluate', {
+        expression: line.toString(),
+        generatePreview: true,
+        throwOnSideEffect: true,
+        timeout: 500,
+        executionContextId: eagerEvalContextId
+      }, (error, previewResult) => {
+
+        if (error) {
+          debug(`Error while generating preview ${error}`);
+          return;
+        }
+
+        if (undefined !== previewResult.result.value) {
+          const value = util.inspect(previewResult.result.value);
+          Interface.prototype._appendPreview.call(this, value);
+          return;
+        }
+
+
+        // If no exception and we have objectId
+        // Run the expression via callFunctionOn
+        // And return it from util inspect.
+        if (!previewResult.exceptionDetails && previewResult.result.objectId) {
+          eagerSession.post('Runtime.callFunctionOn', {
+            functionDeclaration:
+              'function(arg) { return util.inspect(arg, {depth: Infinity}) }',
```

Comment:
`depth: Infinity` does not sound like a good default to me. The preview should ideally be short and on the same line. This could cause the terminal to add lots of output.